### ISSUE TITLE
google play services mismatch with firebase requirement

### ISSUE THREAD

**Issue by rayj10** [Agontuk/react-native-geolocation-service#35]
I added the following dep because I'm still figuring out how to add project wide properties (always causing me more error that I barely understand):

``` 
compile(project(':react-native-geolocation-service')) {
        exclude group: 'com.google.android.gms', module: 'play-services-location'
    }
    compile 'com.google.android.gms:play-services-location:15.0.1'
```
and I used 15.0.1 because I'm also using firebase messaging which requires me to put this:

```
implementation "com.google.android.gms:play-services-base:15.0.1"
    implementation "com.google.firebase:firebase-core:16.0.1"
    implementation "com.google.firebase:firebase-messaging:17.1.0"
    implementation 'me.leolin:ShortcutBadger:1.1.21@aar'
```
in dependencies

and this on the bottom of app level gradle:

```
apply plugin: 'com.google.gms.google-services'
```

with that, I'm getting this error:

```
The library com.google.android.gms:play-services-basement is being requested by various other libraries at [[15.0.1,15.0.1]], but resolves to 11.0.0. Disable the plugin and check your dependencies tree using ./gradlew :app:dependencies.
```

Do u have any suggestion as to what kind of config that would let ur lib to work with this situation?
I'm pretty new with this (managing dependencies manually), so any help would be great 😄 

Thanks in advance for trying to solve RN navigator.geolocation issue :)

**Comment by Agontuk** [comment]
You have to use same version for each of those libraries. Try using 12.0.1 for all of them, 15+ probably won't work if gradle tools version is lower than 3.3 I think.

**Comment by rayj10** [comment]
Thanks for the prompt reply! and I tried just deleting the apply plugin part, it got ur lib to work but not sure  if it affects the firebase. The thing is firebase specifically ask for 15.0.1 but in some forum it has been known that the apply plugin line could cause problems. I just have to test it first if both ur lib and firebase could work using the current setup properly. If not I'll try ur solution using 12.0.1 and report back.

Thanks for the lib! It really solved the problem with RN's geoloc and saved me lots of time :smile:

**Comment by oakis** [comment]
I use 12.0.1 for all libs and that works for me.

**Comment by nickarora** [comment]
You can continue to use 15.0.1 along with react-native-firebase; just make sure you configure your project-wide gradle configuration properties and update `app/build.gradle` appropriately.

ex:
```
/**
 + Project-wide Gradle configuration properties
 */
ext {
    compileSdkVersion   = 25
    targetSdkVersion    = 25
    buildToolsVersion   = "25.0.2"
    supportLibVersion   = "25.0.1"
    googlePlayServicesVersion = "15.0.1"
}
```

**Comment by rayj10** [comment]
@oakis U mean it works for both this lib and firebase messaging like the one I'm trying to use?
Because I'm aware that this lib can work with anything 11+ but the problem is if the firebase part of my code will since they don't really mention it in the docs

@nickarora sounds good I will try adding that partto my project level gradle. I tried adding ext {} yesterday but it gave me error, I think my problem is I'm not sure what to look for in my app level gradle to "update properly" as u mentioned. Sorry I'm pretty new to this, this is the first project that I have to do these stuff manually for RN. :sweat_smile: 

Thanks for the pointers everyone! :smile:

**Comment by oakis** [comment]
@rayj10 Yes i use react-native-firebase, but not with messaging yet. This is part of my build.gradle:

```
implementation "com.google.android.gms:play-services-base:12.0.1"
implementation "com.google.firebase:firebase-core:12.0.1"
implementation "com.google.firebase:firebase-auth:12.0.1"
implementation "com.google.firebase:firebase-database:12.0.1"
implementation 'com.google.android.gms:play-services-location:12.0.1'
```

**Comment by rayj10** [comment]
@oakis cool, I'll try that out then, seems like thats gotta be the optimum solve. Thanks heaps! 😄

**Comment by rayj10** [comment]
So I tried writing my app and project level gradle like the following (based on everyone's recommendations combined):

app level
```
dependencies {
//RNGeolocServices
implementation 'com.google.android.gms:play-services-location:15.0.1' //tried commenting this part out based on this lib's doc's but still no luck
implementation project(':react-native-geolocation-service')
//firebase
implementation "com.google.android.gms:play-services-base:15.0.1"
implementation "com.google.firebase:firebase-core:16.0.1"
implementation "com.google.firebase:firebase-messaging:17.1.0"
implementation 'me.leolin:ShortcutBadger:1.1.21@aar'
```

project level
```
ext {
    compileSdkVersion   = 27
    targetSdkVersion    = 27
    buildToolsVersion   = "27.0.3"
    supportLibVersion   = "25.0.1"
    googlePlayServicesVersion = "15.0.1"
}
```

It solved the original error but giving me this error upon accessing device location:

`{message: "Google play service is not available.", code: 4}`

No idea what I should try now..
thanks in advance guys for the help and patience with my lack of experience 😅

**Comment by Agontuk** [comment]
You have to turn on GPS in your emulator, although testing on emulator won't do you any good since it'll give you a mock location. You have to test on a real device.

**Comment by oakis** [comment]
You can use the emulator as well, just put in coordinates manually and press send. Then test your app and it should receive the location.

**Comment by rayj10** [comment]
@Agontuk So the one returned on emulator will always be mock regardless if I put allow mock on the settings?
I assume the mock property is just checking whether or not allow mock setting is turned on then?

@oakis I'm receiving the location just fine after I fixed the above problem, the problem was apparently I need to put the third argument for the `getCurrentPosition()` that has these { enableHighAccuracy: true, timeout: 2000, maximumAge: 3600000 }. Not sure if the success is dependent on `enableHighAccuracy` being true as some other forum posts said...

Thanks for the amazing support and prompt replies guys! 😄

**Comment by Agontuk** [comment]
No, I meant emulator will not give you the actual location. It'll always return the same position that you entered. If you want to test your actual current position, you're gonna need a real device.
